app/controllers/choices_controller.rb

```diff
@@ -0,0 +1,17 @@
+class ChoicesController < ApplicationController
+
+  def create
+    @question = Question.find(params[:question])
+    text = params[:text]
+    correct = params[:correct] == '1'
+    Choice.create(:text => text, :correct => correct, :question_id => @question.id)
+    redirect_to question_path(@question)
+  end
+
+  def destroy
+    @question = Question.find(params[:question])
+    @choice = Choice.find(params[:choice])
```

Comment:
shouldn't this be params[:id] ?

Reply:
I am passing id in :choice here https://github.com/praneethm381/Questionnaire/pull/1/files#diff-e6ffad0524f0950ee6018b3f68e43342R17

Reply:
the restful url provides us id, e.g. 'choice/:id'

Reply:
this doc will help you understand how nested resources work in rails